### ISSUE TITLE
Misleading information in Readme section Authentication

### ISSUE THREAD

**Issue by LudwigStumpp** [google-github-actions/deploy-cloud-functions#306]
### TL;DR

According to the section `Authentication` of the Readme, one requires authentication to a service account that is a user of the `App Engine Default Service Account`. However after reading [the docs](https://cloud.google.com/functions/docs/reference/iam/roles#additional-configuration) and trying it out on my own, it seems like it suffices to be a member of the `Cloud Function runtime service account` which might be a lot better in terms of security.

### Additional information

The cloud function runtime service account is the account that is set as the attached service account for the cloud function we want to deploy, which is currently specified by the `service_account_email` parameter on the github action.

Declaring a service account as a user of the App Engine Default Service Account might be considered a bad security practise since this allows to basically use all services that are enabled on the Google Cloud Project. By only being a user of the service account that runs the cloud function (and only has the rights it needs), one can minimize the potential harm any authentication theft might have.

**Comment by sethvargo** [comment]
There's a difference between having permissions to _deploy_ a Cloud Function and having permissions to _act as_ a Cloud Function.
